web-app/js/portal/cart/Downloader.js

```diff
@@ -113,13 +113,25 @@ Portal.cart.Downloader = Ext.extend(Ext.util.Observable, {
         });
     },
 
-    _onAsyncDownloadRequestSuccess: function() {
+    _onAsyncDownloadRequestSuccess: function(response) {
+        var serviceMessage = "";
+        if (this.params.serviceResponseHandler) {
+            serviceMessage = this.params.serviceResponseHandler(response.responseText);
+        }
```

Comment:
Na, this is fine now. I can break it to another function if you really want me to, however it'll entail even bit more changes here and there as the scope passed to this function is `params` rather than `this` as the Downloader.